Mathlib/RepresentationTheory/Rep.lean

```diff
@@ -136,6 +136,18 @@ theorem trivial_def {V : Type u} [AddCommGroup V] [Module k V] (g : G) (v : V) :
 set_option linter.uppercaseLean3 false in
 #align Rep.trivial_def Rep.trivial_def
 
+/-- A predicate for representations that fix every element. -/
+class isTrivial (A : Rep k G) : Prop :=
+(out : ∀ g x, A.ρ g x = x)
```

Comment:
The more standard Mathlib4 syntax would be
```suggestion
class IsTrivial (A : Rep k G) : Prop where
  out : ∀ g x, A.ρ g x = x
```
In my understanding, `IsTrivial` should be capitalized.

Then, in order to use the assumption, instead of given a name like `[h : ρ.isTrivial]`, you should define a simp lemma `Rep.ρ_apply_eq_self` (or something like that).

Also, you may try to declare this as `  out : ∀ g x, A.ρ g x = x := by aesop` instead so that the next instances could be defined just as
`instance {V : Type u} [AddCommGroup V] [Module k V] : IsTrivial (Rep.trivial k G V) where`
where the proof would be automatically found by aesop (and if aesop does not, add suitable simp lemmas until it does!).

(Same remarks should also apply to the similar definition in `Basic.lean.`)

Reply:
Thank you! Done, and made `Rep.IsTrivial A` an abbreviation for `Representation.IsTrivial A.ρ` so we just need one simp lemma.